bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/connection/ValloxBaseConnector.java

```diff
@@ -0,0 +1,359 @@
+                    } else {
+                        waitForAckByte.set(true);
+                        writeToOutputStream(telegram);
+                        sendQueue.addFirst(queueItem);
+                    }
+                } else {
+                    sendErrorToListeners("Ack byte not received for telegram: " + telegram.toString(), null);
+                }
+                break;
+            default:
+                logger.debug("Unknown telegram in send queue: {}", telegram.state);
+                break;
+        }
+    }
+
+    /**
+     * Write telegram bytes to output stream
+     *
+     * @param telegram the telegram to write
+     */
+    private void writeToOutputStream(Telegram telegram) {
+        OutputStream outputStream = this.outputStream;
+        if (outputStream != null) {
+            try {
+                outputStream.write(telegram.bytes);
+                outputStream.flush();
+                logger.debug("Wrote {}", telegram);
+            } catch (IOException e) {
+                sendErrorToListeners("Write to output stream failed, " + e.getMessage(), e);
+                sendQueue.clear();
+            }
+        } else {
+            logger.debug("Output stream is null");
+        }
+    }
+
+    /**
+     * {@link Thread} implementation for processing telegrams
+     *
+     * @author Miika Jukka - Initial contribution
+     */
+    private class TelegramProcessor extends Thread {
+        boolean interrupted = false;
+
+        public TelegramProcessor(String name) {
+            super(name);
+            setDaemon(true);
+        }
+
+        @Override
+        public void interrupt() {
+            interrupted = true;
+            super.interrupt();
+        }
+
+        @Override
+        public void run() {
+            logger.trace("Telegram processor thread started");
+            while (!interrupted) {
+                try {
+                    handleBuffer();
+                } catch (InterruptedException e) {
+                    sendErrorToListeners("Buffer handling interrupted", e);
+                    interrupt();
+                }
+            }
+            logger.trace("Telegram processor thread stopped");
+        }
+    }
+
+    /**
+     * Handle send queue in own thread
+     *
+     * @author Miika Jukka - Initial contribution
+     */
+    private class SendQueueHandler extends Thread {
+        boolean interrupted = false;
+
+        public SendQueueHandler(String name) {
+            super(name);
+            setDaemon(true);
+        }
+
+        @Override
+        public void interrupt() {
+            interrupted = true;
+            super.interrupt();
+        }
+
+        @Override
+        public void run() {
+            logger.trace("Send queue handler thread started");
+            while (!interrupted) {
+                try {
+                    handleSendQueue();
+                    sleep(500);
```

Comment:
Could you make this a periodic scheduled task instead?

Reply:
It was already changed when you commented. It showed the `Outdated` label as it is showing now. 🤔

Reply:
I must have been reviewing and older version of the code.  No biggie, just disregard this change request then!